snapcraft/tests/plugins/test_nodejs.py

```diff
@@ -291,6 +298,50 @@ class Options:
         self.assertFalse(os.path.exists(plugin._npm_dir))
 
 
+class NodePluginManifestTestCase(NodePluginBaseTestCase):
+
+    scenarios = multiply_scenarios(
+        [
+            ('simple', dict(return_value=(
+                '{"dependencies": {'
+                '   "testpackage1": {"version": "1.0"},'
+                '   "testpackage2": {"version": "1.2"}}}'))),
+            ('nested', dict(return_value=(
+                '{"dependencies": {'
+                '   "testpackage1": {'
+                '      "version": "1.0",'
+                '      "dependencies": {'
+                '        "testpackage2": {"version": "1.2"}}}}}'))),
+            ('missing', dict(return_value=(
+                '{"dependencies": {'
+                '   "testpackage1": {"version": "1.0"},'
+                '   "testpackage2": {"version": "1.2"},'
+                '   "missing": {"noversion": "dummy"}}}')))],
+        [('npm', dict(package_manager='npm')),
+         ('yarn', dict(package_manager='yarn'))])
+
+    def test_get_manifest_with_node_packages(self):
+        self.run_output_mock.return_value = self.return_value
```

Comment:
Can you call the variable from the scenarios dict `dependencies` for clarity?

Reply:
I don't think that would be clearer. Nodejs is weird in that ls returns a dictionary. Inside that dictionary, there is a key called "dependencies". But it could have other keys.
This argument is not the dependencies, it's the string value that is returned by run_output.

Reply:
`package_list` maybe? Anything's better than return_value...

Reply:
I renamed it to ls_output.